File: fix_markdown.py
import re

with open('src/components/InfoPage.tsx', 'r') as f:
    content = f.read()

# Isolate the markdown content
parts = content.split('const markdownContent = `')
if len(parts) == 2:
    header = parts[0]
    rest = parts[1]
    
    md_parts = rest.split('`;\n\nexport function InfoPage() {')
    md = md_parts[0]
    footer = '`;\n\nexport function InfoPage() {' + md_parts[1]
    
    # 1. Unescape double backslashes for Math macros since we will use String.raw
    md = md.replace('\\\\', '\\')
    
    # 2. Fix the issues with KaTeX choking on \$ inside math mode
    md = md.replace('$LR < \\$100,000$', '$LR < 100,000$')
    md = md.replace('$LR = \\$33,000$', '$LR = 33,000$')
    md = md.replace('$PV = \\$100,000$', '$PV = 100,000$')
    md = md.replace('$C = \\$10,000\\text{/year}$', '$C = 10,000\\text{/year}$')
    md = md.replace('First income = $50,000 \\times 1.03^{20} = \\$90,305$.', 'First income = $\\text{\\$50,000} \\times 1.03^{20} = \\text{\\$90,305}$.')
    md = md.replace('$FV = 90,305 \\times \\frac{1 - 1.0506^{-30}}{0.0506} \\approx \\$1,386,000$.', '$FV = 90,305 \\times \\frac{1 - 1.0506^{-30}}{0.0506} \\approx 1,386,000$.')
    md = md.replace('$PV = \\$20,000$', '$PV = 20,000$')
    md = md.replace('$C = \\$5,000$', '$C = 5,000$')
    md = md.replace('target = \\$200,000$ (nominal)', 'target = \\$200,000 (nominal)')
    
    # 3. Fix unescaped dollar signs in text that cause math block matching issues
    md = md.replace('Cash: $8,000, Bonds: $25,000', 'Cash: \\$8,000, Bonds: \\$25,000')
    md = md.replace('Long‑horizon equity (goals >10yr) = $200,000 → > $250k?', 'Long‑horizon equity (goals >10yr) = \\$200,000 → > \\$250k?')
    
    # Reassemble with String.raw
    new_content = header + 'const markdownContent = String.raw`' + md + footer
    
    with open('src/components/InfoPage.tsx', 'w') as f:
        f.write(new_content)
    print("Done")
else:
    print("Failed to split")

File: fix_markdown.js
const fs = require('fs');
let content = fs.readFileSync('src/components/InfoPage.tsx', 'utf8');

// Replace template literal start
content = content.replace('const markdownContent = `', 'const markdownContent = String.raw`');

// Fix unescaped dollar signs in text that cause math block matching issues
content = content.replace('Cash: $8,000, Bonds: $25,000', 'Cash: \\$8,000, Bonds: \\$25,000');
content = content.replace('Long‑horizon equity (goals >10yr) = $200,000 → > $250k?', 'Long‑horizon equity (goals >10yr) = \\$200,000 → > \\$250k?');

// Fix dollar signs inside math mode
content = content.replace('$LR < \\\\$100,000$', '$LR < 100,000$');
content = content.replace('$LR = \\\\$33,000$', '$LR = 33,000$');
content = content.replace('$PV = \\\\$100,000$, $C = \\\\$10,000\\\\text{/year}$', '$\\text{PV} = 100,000$, $\\text{C} = 10,000\\text{/year}$');
content = content.replace('Desired income \\\\$50,000', 'Desired income \\$50,000');
content = content.replace('$50,000 \\\\times', '50,000 \\times'); // wait, First income = $50,000 \times 1.03^{20} = \$90,305$
// Wait, the First income line is:
// First income = $50,000 \\times 1.03^{20} = \\$90,305$.
// We want it to be:
// First income = $50,000 \times 1.03^{20} = 90,305$.
content = content.replace('First income = $50,000 \\\\times 1.03^{20} = \\\\$90,305$.', 'First income = $\\$50,000 \\times 1.03^{20} = \\$90,305$.'); // Wait, if I do this, kaTeX will fail on \$!
// Let's just make it outside math!
content = content.replace('First income = $50,000 \\\\times 1.03^{20} = \\\\$90,305$.', 'First income = $\\text{\\$50,000} \\times 1.03^{20} = \\text{\\$90,305}$.');
content = content.replace('$FV = 90,305 \\\\times \\\\frac{1 - 1.0506^{-30}}{0.0506} \\\\approx \\\\$1,386,000$.', '$FV = 90,305 \\times \\frac{1 - 1.0506^{-30}}{0.0506} \\approx 1,386,000$.');
content = content.replace('$PV = \\\\$20,000$, $C = \\\\$5,000$, $T=8$, target = \\\\$200,000$ (nominal).', '$\\text{PV} = 20,000$, $\\text{C} = 5,000$, $T=8$, target = \\$200,000 (nominal).');

// Wait, since I'm switching to String.raw, any existing \\ something will become \\ something!
// E.g. \\times will become literally \ \ t i m e s!
// I need to change all \\ to \ in the markdownContent string!
